crypto/pkcs7/pkcs7.c

```diff
@@ -1548,26 +1575,65 @@ static int pkcs7_signature_verify(BIO *in_bio, PKCS7 *p7, PKCS7_SIGNER_INFO *si,
     bio = BIO_next(bio);
   }
 
-  // We don't currently support signed attributes. See |PKCS7_NOATTR|.
-  if (si->auth_attr && sk_X509_ATTRIBUTE_num(si->auth_attr) != 0) {
-    OPENSSL_PUT_ERROR(PKCS7, PKCS7_R_INVALID_SIGNED_DATA_TYPE);
+  // mdc is the digest ctx that we want, unless there are attributes, in
+  // which case the digest is the signed attributes.
+  if (!EVP_MD_CTX_copy_ex(mdc_tmp, mdc)) {
     goto out;
```

Comment:
`mdc` is initialized to NULL on L1559, never populated, then used as the `in` parameter in the call to `EVP_MD_CTX_copy_ex`. so what are we copying into `mdc_tmp` here? and why allocate to `mdc_tmp` if it just gets overwritten here before it's used?

Reply:
`mdc` is initially `NULL`, until we find the correct `EVP_MD_CTX` to work with from `bio`. That is then retrieved with `BIO_get_md_ctx` on L1568, which directly returns a pointer to the underlying `EVP_MD_CTX`.

I believe the copying's done because upstream/we wanted to use `EVP_MD_CTX` for further verification operations, but we still wished to preserve the original state of the retrieved context. This allows for multiple operations of `PKCS7_verify`, without consuming the context. Perhaps we should add a test for that as well.

Reply:
Test added!